Formula/vim.rb

```diff
@@ -42,6 +42,7 @@ class Vim < Formula
   depends_on "lua" => :optional
   depends_on "luajit" => :optional
   depends_on :x11 if build.with? "client-server"
+  depends_on "gettext" if build.with? "nls"
```

Comment:
Asking because I made a stink about that over here: https://github.com/Homebrew/homebrew-dupes/pull/680

So if we've decided `"with-nls"` is acceptable after all, I don't want to be inconsistent.